Automattic/jetpack
# Lazy Images: Use calypso-build to build

Fixes #17562 -- or rather, most of it:

> * us[e] calypso-build (which takes care of transpilation, and by default drops files into a separate build/ [edit: should be `dist/`] directory, thus resolving items 1. and 2.)
> * delet[e] built files from GH, and adding the [`dist/`] subdirectory to .gitignore in order to prevent that from happening in the future
> * Revert https://github.com/Automattic/jetpack/pull/17561 to re-enable validation.

One item from #17562 isn't addressed by this PR:

> * add the Interaction Observer npm to package.json (rather than copying it verbatim)

I haven't yet found a really good way to do that, as we also need to build it, and include it as an external. We can address that in a subsequent iteration however.

#### Changes proposed in this Pull Request:
Use `calypso-build` to build the lazy-images package.

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

- Run `yarn build-production-packages`, and verify that it passes (it includes ES5 validation!).
- Verify that the Lazy images module still works (specifically, the built JS).

(I've tested a post with an image and the Lazy Images setting switched on and it seemed to work fine, but I'm not familiar enough with lazy images to be 100% certain that it's still working as before.)

#### Proposed changelog entry for your changes:
Use `calypso-build` to build the lazy-images package.


#### Note

In comments on #17562, it was brought up that we might need to include the built `js` files in GH in order for composer to be able to export this package. This would mean adding the `dist/` directory. Should we do that? Or can composer include a build step for JS?
